protocol/test/crypto-primitives.ts

```diff
@@ -0,0 +1,54 @@
+import { expect } from "chai";
+const { ethers } = require("hardhat");
+
+import { CardsStructs } from "../typechain-types/crypt/DeckCrypt";
+import DeckStruct = CardsStructs.DeckStruct;
+
+// Deck encrytion and decryption routing mirrors the one in the contract DeckCrypt.sol
+export function encryptUsingOTP(
+  deck: DeckStruct,
+  key: Array<string>
+): DeckStruct {
+  expect(key.length).to.be.greaterThan(deck.cards.length);
+
+  const encryptedDeck: DeckStruct = {
+    cards: [],
+  };
+
+  for (let i = 0; i < deck.cards.length; i++) {
+    let card = deck.cards[i];
+
+    // Do XOR byte by byte
+    let xored = parseInt(card.toString(), 10) ^ parseInt("0x" + key[i], 16);
+
+    // Add padding zeroes to the left to make sure the length is 2
+    xored = ("0" + xored.toString(16)).substr(-2);
+
+    encryptedDeck.cards[i] = "0x" + xored;
+  }
+  return encryptedDeck;
+}
+
+// Generate secret key for OTP encryption using keccak256 mirroring the contract implementation
+export function generateKeyUsingSecret(
+  secret: string,
+  length: number
+): Array<string> {
+  expect(length).to.be.greaterThan(0);
+
+  let sequence: number = 0;
+  let key: string = "";
+
+  for (let i = 0; i < length / 32; i++) {
```

Comment:
Se podría iterar sobre sequence en lugar de sobre i, no?

Reply:
`i` represents current bytes[] of key material. `sequence` was brought in to this because keccak expands on 32 bytes basis, so that's keeps states of how many "key expansions" were performed